data/data/bootstrap/files/usr/local/bin/installer-gather.sh

```diff
@@ -71,53 +76,79 @@ function queue() {
+queue resources/rolebindings.json oc --request-timeout=5s get rolebindings --all-namespaces -o json
+queue resources/roles.json oc --request-timeout=5s get roles --all-namespaces -o json
 # this just lists names and number of keys
-queue resources/secrets-names.txt oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get secrets --all-namespaces
+queue resources/secrets-names.txt oc --request-timeout=5s get secrets --all-namespaces
 # this adds annotations, but strips out the SA tokens and dockercfg secrets which are noisy and may contain secrets in the annotations
-queue resources/secrets-names-with-annotations.txt oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get secrets --all-namespaces -o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name,TYPE:.type,ANNOTATIONS:.metadata.annotations | grep -v -- '-token-' | grep -v -- '-dockercfg-'
-queue resources/services.json oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get services --all-namespaces -o json
+queue resources/secrets-names-with-annotations.txt oc --request-timeout=5s get secrets --all-namespaces -o=custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name,TYPE:.type,ANNOTATIONS:.metadata.annotations | grep -v -- '-token-' | grep -v -- '-dockercfg-'
+queue resources/services.json oc --request-timeout=5s get services --all-namespaces -o json
 
-FILTER=gzip queue resources/openapi.json.gz oc --kubeconfig=/opt/openshift/auth/kubeconfig --request-timeout=5s get --raw /openapi/v2
+FILTER=gzip queue resources/openapi.json.gz oc --request-timeout=5s get --raw /openapi/v2
 
 echo "Waiting for logs ..."
 wait
 
+# The existence of the file located in LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME is used
+# as indication that a bootstrap process has already previously taken place and the resulting
+# bundle already exists in the filesystem. In that case, we replace some of the logs we've
+# just collected with the logs from said bundle.
+#
+# An example for a scenario where this happens is when we're running inside a single-node
+# bootstrap-in-place deployment post-pivot master node, rather than a typical bootstrap node.
+# In that case, the bootstrap node collects and bundles logs pre-reboot and transfers that bundle
+# via an ignition file to this post-pivot master node.
+LOG_BUNDLE_BOOTSTRAP_NAME="log-bundle-bootstrap"
+LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME="/var/log/${LOG_BUNDLE_BOOTSTRAP_NAME}.tar.gz"
+
+if [[ -f ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME} ]]; then
+    echo "Including existing bootstrap bundle ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME}"
+    tar -xzf ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME}
+    mkdir -p "${ARTIFACTS}/bootstrap-log-bundle"
+    cp -r "${LOG_BUNDLE_BOOTSTRAP_NAME}"/* "${ARTIFACTS}/bootstrap-log-bundle/"
+fi
+
 echo "Gather remote logs"
 export MASTERS=()
-if [ "$#" -ne 0 ]; then
+if [[ -f ${LOG_BUNDLE_BOOTSTRAP_ARCHIVE_NAME} ]]; then
```

Comment:
is it worth switching on this dynamically at script-run-time, instead of making this `installer-gather.sh.template` and switching on `{{- if .BootstrapInPlace}}` at script-creation-time?

Reply:
The same script is executed on the bootstrap prior to the reboot (where the log bundle doesn't exists) and after the reboot (where we need it to add the bootstrap log bundle).
This is why we are switching on run-time.